R/ParamTree.R

```diff
@@ -0,0 +1,45 @@
+#' @title ParamTree
+#' @format \code{\link{R6Class}} object
+#'
+#' @description
+#' A \code{\link[R6]{R6Class}} to represent set of parameters in a tree form.
+#'
+#' @return [\code{\link{ParamTree}}].
+#' @family ParamHelpers
+#' @export
+ParamTree = R6Class("ParamSetTree",
+  inherit = ParamSet,
+  public = list(
+
+    # member variables
+    id = NULL,
+    handle = NULL,
+    flatval = NULL,
+    depend = NULL,
+
+    # constructor
+    initialize = function(id = NULL, handle = NULL, depend = NULL) {
+      self$id = id
+      self$handle = handle
+      self$flatval = list()
+      self$depend = depend
+    },
+
+    # public methods
+    sample = function() {
+      self$handle$sample()
+    }
+  ),
+  private = list(
+  )
+)
+ParamTree$fac = function(...) {
```

Comment:
Imho we should just use normal functions. There is no need to put this function in that class object. Furthermore it is not documented in R6 so we can not be sure that this will work in the future.

Reply:
so the solution is to replace this ParamTree$fact to a normal function `ParamTreeFac` but still put this normal function `ParamTreeFac` in the same file ?

Reply:
Yes. But You could also make a file ParamTree_functions.R - maybe?